app/lib/foreman_webhooks/renderer/scope/webhook_template.rb

```diff
@@ -21,14 +21,15 @@ def initialize(**args)
         end
 
         apipie :method, 'Creates final payload' do
-          required :hash, Hash, 'Key=value object with with data that should be present in payload'
+          optional :hash, Hash, 'Key=value object with with data that should be present in payload', default: {}
           keyword :with_defaults, [true, false], 'If set to true, adds default entries to the payload', default: true
           returns String, 'JSON string with the final payload'
           example 'payload({ id: @object.id, name: @object.name }) #=> ' \
                   '"{ "id": 1, "name": "host.example.com", "context": { ... }, ' \
                   '"event_name": "host_created.event.foreman" }"'
         end
-        def payload(hash, with_defaults: true)
+        def payload(hash = {}, with_defaults: true, **kwargs)
+          hash.merge!(kwargs)
```

Comment:
Could we only do this if the kwargs are not empty?

An edge case that worked before, but wouldn't work anymore would be something like this
```ruby
payload([1, 2, 3], with_defaults: false)
```

Reply:
Even though it could work before, it would be unsupported usage?

We could add here a clause to check if the provided values for this method are hash and/or keywords and raise an error if it's not. WDYT?

Reply:
> it would be unsupported usage?

Out of curiosity, why?

> We could add here a clause to check if the provided values for this method are hash and/or keywords and raise an error if it's not. WDYT?

If passing something else in would make it blow up in some random place somewhere down the road, then yes it would be better to catch it here

Reply:
> Out of curiosity, why?

We already had `hash.merge!(@defaults) if with_defaults` with `with_defaults` being `true` by default, so if one passes an array it would raise an error.

> If passing something else in would make it blow up in some random place somewhere down the road, then yes it would be better to catch it here

Added a simple check.